ui/ducks/send/send.js

```diff
@@ -723,6 +744,72 @@ export const initializeSendState = createAsyncThunk(
   },
 );
 
+/**
+ * Fetch the swap and send transaction if the source and destination token do not match
+ *
+ * @param {string} hexData - hex encoded string representing transaction data.
+ * @returns {ThunkAction<void>}
+ */
+
+export const fetchSwapAndSendQuotes = createAsyncThunk(
+  'send/fetchSwapAndSendQuotes',
+  async (_, thunkApi) => {
+    const state = thunkApi.getState();
+    const sendState = state[name];
+
+    const chainId = getCurrentChainId(state);
+
+    const draftTransaction =
+      sendState.draftTransactions[sendState.currentTransactionUUID];
+
+    // TODO: move to util
+    const sender =
+      draftTransaction.fromAccount?.address ??
+      sendState.selectedAccount.address ??
+      getSelectedInternalAccount(state).address;
+
+    // return early if swap isn't required
+    if (
+      draftTransaction?.sendAsset?.details?.address ===
+      draftTransaction?.receiveAsset?.details?.address
+    ) {
+      return { quotes: null };
+    }
+
+    const sourceAmount = hexToDecimal(draftTransaction.amount.value);
+
+    // return early if form isn't filled out
+    if (
+      !Number(sourceAmount) ||
+      // !draftTransaction.sendAsset ||
+      // !draftTransaction.receiveAsset ||
+      !draftTransaction.recipient.address
+    ) {
+      return { quotes: null };
+    }
+
+    const quotes = await getSwapAndSendQuotes({
+      chainId,
+      sourceAmount,
+      sourceToken:
+        draftTransaction.sendAsset?.details?.address ||
+        '0x0000000000000000000000000000000000000000',
+      destinationToken:
+        draftTransaction.receiveAsset?.details?.address ||
+        '0x0000000000000000000000000000000000000000',
+      sender,
+      recipient: draftTransaction.recipient.address,
+      slippage: '5', // TODO: update when solution is available
+    });
+
+    if (!Object.keys(quotes).length) {
+      throw new Error(SWAPS_QUOTES_ERROR);
+    }
+
+    return { quotes };
```

Comment:
```suggestion
    try {
      const quotes = await getSwapAndSendQuotes({
        chainId,
        sourceAmount,
        sourceToken:
          draftTransaction.sendAsset?.details?.address ||
          '0x0000000000000000000000000000000000000000',
        destinationToken:
          draftTransaction.receiveAsset?.details?.address ||
          '0x0000000000000000000000000000000000000000',
        sender,
        recipient: draftTransaction.recipient.address,
        slippage: '5', // TODO: update when solution is available
      });

      if (!Object.keys(quotes).length) {
        throw new Error(QUOTES_NOT_AVAILABLE_ERROR);
      }

      return { quotes };
    } catch () {
      throw new Error(SWAPS_QUOTES_ERROR);
    }
```

Reply:
Should we have a different error state when there are no quotes? Also suggest throwing the generic quotes error in a try/catch block

Reply:
Good call – I used `.catch()` instead of try-catch